services/rds/parameter_group_test.go

```diff
@@ -91,15 +155,256 @@ func TestNeedCustomParameters(t *testing.T) {
+							},
+						},
+					},
+				},
+			},
+			expectedParamValue: "random-library",
+		},
+		"default param value, with paging": {
+			dbInstance: &RDSInstance{
+				EnablePgCron: true,
+				DbType:       "postgres",
+				DbVersion:    "12",
+			},
+			paramName: "shared_preload_libraries",
+			expectedParams: map[string]map[string]string{
+				"postgres": {
+					"shared_preload_libaries": "pg-cron",
+				},
+			},
+			describeEngineDefaultParamsResults: []*rds.DescribeEngineDefaultParametersOutput{
+				{
+					EngineDefaults: &rds.EngineDefaults{
+						Parameters: []*rds.Parameter{
+							{
+								ParameterName:  aws.String("random-param"),
+								ParameterValue: aws.String("random-value"),
+							},
+						},
+						Marker: aws.String("marker"),
+					},
+				},
+				{
+					EngineDefaults: &rds.EngineDefaults{
+						Parameters: []*rds.Parameter{
+							{
+								ParameterName:  aws.String("shared_preload_libraries"),
+								ParameterValue: aws.String("a-library,b-library"),
+							},
+						},
+					},
+				},
+			},
+			expectedParamValue: "a-library,b-library",
+		},
+		"describe db engine params error": {
+			dbInstance: &RDSInstance{
+				EnablePgCron: true,
+				DbType:       "postgres",
+				DbVersion:    "12",
+			},
+			paramName: "shared_preload_libraries",
+			expectedParams: map[string]map[string]string{
+				"postgres": {
+					"shared_preload_libaries": "pg-cron",
+				},
+			},
+			describeEngineDefaultParamsErr: describeEngineDefaultParamsErr,
+			expectedErr:                    describeEngineDefaultParamsErr,
+			expectedParamValue:             "",
+		},
+		"describe db engine versions error": {
+			dbInstance: &RDSInstance{
+				EnablePgCron: true,
+				DbType:       "postgres",
+			},
+			paramName: "shared_preload_libraries",
+			expectedParams: map[string]map[string]string{
+				"postgres": {
+					"shared_preload_libaries": "pg-cron",
+				},
+			},
+			describeEngineDefaultParamsResults: []*rds.DescribeEngineDefaultParametersOutput{
+				{
+					EngineDefaults: &rds.EngineDefaults{
+						Parameters: []*rds.Parameter{},
+					},
+				},
+			},
+			describeEngVersionsErr: describeEngVersionsErr,
+			expectedErr:            describeEngVersionsErr,
+			expectedParamValue:     "",
+		},
+	}
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			describeEngineCallNum = 0
+			paramValue, err := getDefaultEngineParameter(test.paramName, test.dbInstance, &mockRDSClient{
+				describeEngineDefaultParamsResults: test.describeEngineDefaultParamsResults,
+				describeEngineDefaultParamsErr:     test.describeEngineDefaultParamsErr,
+				describeEngVersionsErr:             test.describeEngVersionsErr,
+			})
+			if test.expectedErr == nil && err != nil {
+				t.Errorf("unexpected error: %s", err)
+			}
+			if test.expectedErr != nil && !errors.Is(err, test.expectedErr) {
+				t.Errorf("expected error: %s, got: %s", test.expectedErr, err)
+			}
```

Comment:
No need to check for nil; `errors.Is()` will return true if they're both nil. (Try it yourself: https://go.dev/play/p/MgsKWwoTUKF)

```suggestion
			if !errors.Is(err, test.expectedErr) {
				t.Errorf("expected: %s, got: %s", test.expectedErr, err)
			}
```